app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -313,23 +313,22 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
         }
 
         consumeFrom(store) {
-            val shouldShowAwesomebar =
-                !firstUpdate &&
-                it.query.isNotBlank() ||
-                it.showSearchShortcuts
-
-            awesome_bar?.visibility = if (shouldShowAwesomebar) View.VISIBLE else View.INVISIBLE
+            awesome_bar?.visibility = if (shouldShowAwesomebar(it)) View.VISIBLE else View.INVISIBLE
             updateSearchSuggestionsHintVisibility(it)
             updateClipboardSuggestion(it, requireContext().components.clipboardHandler.url)
             updateToolbarContentDescription(it)
             updateSearchShortcutsIcon(it)
             toolbarView.update(it)
             awesomeBarView.update(it)
-            firstUpdate = false
+            if (it.url != it.query) firstUpdate = false
```

Comment:
It looks like if we move this line on top of this block then we can simplify `shouldShowAwesomebar` as the additional url != query check isn't needed.

```kotlin
consumeFrom(store) {
  if (it.url != it.query) firstUpdate = false
  awesome_bar?.visibility = if (shouldShowAwesomebar(it)) View.VISIBLE else View.INVISIBLE
  ...
}
```

```kotlin
private fun shouldShowAwesomebar(searchFragmentState: SearchFragmentState) =
  !firstUpdate && searchFragmentState.query.isNotBlank() || searchFragmentState.showSearchShortcuts

```

Reply:
Could we also add a comment about what `firstUpdate` is and why we use it?